accepted/0000-retail-pxe-database.md

```diff
@@ -0,0 +1,259 @@
+  
+  * will it impact performance?
+  
+    Multiple terminals booting at the same time - current state is handled solely by salt, proposed solution adds communication with database - increasing latency and some load on database
+  
+  * what other parts of the product will be affected?
+  
+    There should be no affect to other parts of product
+  
+  * will the solution be hard to maintain in the future?
+  
+    As is the case even now, the code is split across different projects and github repositories. Saltboot code in [uyuni-retail/saltboot-formula](https://github.com/uyuni-project/retail/tree/master/saltboot-formula), `pxe/terminal_entry` state in [salt-formulas/pxe-formula](https://github.com/SUSE/salt-formulas/tree/master/pxe-formula) next adding java event handler adds third repository. However code itself is not complex and this code split can be mitigated either by moving parts closer together (like [integrating saltboot to uyuni](https://github.com/SUSE/spacewalk/issues/10777)) or good code commenting.
+  
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Automatic detection of partition where salt configuration is stored
+
+  Draft PR [is here](https://github.com/uyuni-project/retail/pull/71). This removes the need to reregister terminal in case of lost PXE entry. However this does not preserve terminal-specific kernel command line options and always boot default initrd and kernel.
+
+- TPM as storage for salt configuration
+
+  Trusted Platform Module is required to be available on every hardware certified to run MS Windows 10, essentially every new hardware. It can be used as a secure storage for salt keys and minion id instead of storing it on hard drive. This would solve the same problem as automatic detection of salt configuration with the same drawbacks. This would also need some more time for research and packaging ensuring tpm2-tss is available in saltboot initrd.
+
+- Storage of pillar data in pillar files instead of database
+
+  This would allow to reuse of current external pillar mechanism, but suffer from the same issues like existing implementation - maintaining consistency of data and others. (See [overall issue](https://github.com/SUSE/spacewalk/issues/10679))
+
+- Detection of `pxeserver_id` can also be done similarly as `salt` CNAME is being resolved by saltboot. That means adding `tftp` CNAME as hard requirement for DNS resolution. It is already documented to do so, however it wasn't used for anything and given number of users use their DNS servers instead of retail provided one this can introduce a regression in users scenarios.
+
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Is presented detection of `pxeserver_id` acceptable?
+
+- `pxeserver_id` when terminal is not connected to the proxy, but directly to the Uyuni/SUSE Manager is `null` and has a functional meaning. However the same state can be achieved by removing branch server server entry. How big problem is it?
```

Comment:
Detection from DHCP next-server is a good idea.

As I wrote above, a terminal connected directly can be determined more reliably, it does not have to be just a fallback.
Also in some alternatives in Retail Hub RFC this is not needed because Branch server will be allways connected as minion.